### PR TITLE
Feature/reaction questions

### PR DESCRIPTION
**Pull request by moh-moola**
Move reaction question logic from `molo.core`

### CODE REVIEW

File: molo/forms/tests/test_reaction_questions.py [hunk 1]
```diff
@@ -0,0 +1,528 @@
...
+
+        messages = list(get_messages(response.wsgi_request))
+        self.assertEqual(len(messages), 1)
+        # this should show if the submit is not done via ajax
+        self.assertIn(
+            'You have already given feedback on this article.',
+            str(messages[0])
+        )
+
+    def test_can_react_on_article_ajax_call(self):
```
Comment (KaitCrawford) [inline molo/forms/tests/test_reaction_questions.py]: I think this test needs to submit the reaction question response twice (like the one above).
If I'm not mistaken there should be 0 ReactionQuestionResponse objects at the start of this test. The first post to submit a response creates one and then the second post should change the chosen value. Currently you're only creating the first one. So you wouldn't get an error message anyway and you're not testing that the value gets changed on the second submission.

File: molo/forms/tests/test_reaction_questions.py [hunk 2]
```diff
@@ -0,0 +1,528 @@
...
+
+        res = self.client.get(url)
+        self.assertTrue(res.status_code, 200)
+        self.assertEqual(
+            res.template_name,
+            ['forms/reaction_question_feedback.html'])
+        self.assertContains(res, article.title)
+        self.assertContains(res, choice.success_message)
+
+    def test_load_user_choice_reaction_question(self):
```
Comment (KaitCrawford) [inline molo/forms/tests/test_reaction_questions.py]: This test and the one below it aren't related to the ReactionQuestionChoiceFeedbackView are they?
Maybe they should be in their own class?

File: molo/forms/tests/test_reaction_questions.py [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+            language_setting=Languages.for_site(self.main.get_site()),
+            locale='en',
+            is_active=True)
+
+        self.assertTrue(self.section_index)
+        self.yourmind = self.mk_section(
+            self.section_index, title='Your mind')
+        self.reaction_index = self.get_reaction_index(self.main)
+
+    def test_view(self):
```
Comment (KaitCrawford) [inline molo/forms/tests/test_reaction_questions.py]: Please add a more detailed name or comment here. What is the view that you're testing expected to do?

File: molo/forms/wagtail_hooks.py [hunk 4]
```diff
@@ -47,7 +45,22 @@ def create_new_page_relations(request, page, new_page):
...
-# This overrwrites the wagtailsuveys admin urls in order to use custom
+@hooks.register('construct_main_menu')
+def show_reactionquestions_response_for_users_have_access(request, menu_items):
+    excl = [
+        'reaction-question',
+        'reaction-question-responses',
+        'reaction-question-summary',
+    ]
+    menu_items[:] = [
+        item for item in menu_items if item.name not in excl]
```
Comment (KaitCrawford) [inline molo/forms/wagtail_hooks.py]: What is the purpose of this section above?

**Review by KaitCrawford — COMMENTED** [review]
@moh-moola I've left a few comments

File: molo/forms/wagtail_hooks.py (hunk 4 above)
Comment (moh-moola) [reply molo/forms/wagtail_hooks.py]: @KaitCrawford remove the menu Items in the list from the main menu since they are subs of `reactionquestions`

Comment (KaitCrawford) [reply molo/forms/wagtail_hooks.py]: Do you mean this menu?
Without this code do they show up in the left column or the right column?
cms_menu

File: molo/forms/tests/test_reaction_questions.py [hunk 5]
```diff
@@ -0,0 +1,554 @@
...
+            self.assertContains(response, choice.title)
+
+        choice2 = question.get_children().last()
+        # submit with ajax in post data
+        response = self.client.post(
+            reverse('molo.forms:reaction-vote', kwargs=kw),
+            {'choice': choice2.id, 'ajax': 'True'})
+
+        # there should still only be one response object from user
+        # but the vote should change from choice 1 to choice 2
```
Comment (KaitCrawford) [inline molo/forms/tests/test_reaction_questions.py]: This comment belongs further down, it's referencing the second submission

File: molo/forms/tests/test_reaction_questions.py [hunk 6]
```diff
@@ -0,0 +1,554 @@
...
+        self.assertEqual(ReactionQuestionResponse.objects.all().count(), 1)
+        self.assertEqual(
+            ReactionQuestionResponse.objects.last().choice.pk, choice2.pk)
+        messages = list(get_messages(response.wsgi_request))
+        # no error message if the submit is done via ajax
+        self.assertEqual(len(messages), 0)
+
+        response = self.client.post(
+            reverse('molo.forms:reaction-vote', kwargs=kw),
+            {'choice': choice2.id, 'ajax': 'True'})
```
Comment (KaitCrawford) [inline molo/forms/tests/test_reaction_questions.py]: Please change the choice here. We're trying to test that if the user changes their choice it doesn't make a new response object but overwrites the old one

File: molo/forms/wagtail_hooks.py (hunk 4 above)
Comment (moh-moola) [reply molo/forms/wagtail_hooks.py]: @KaitCrawford yes

**Review by KaitCrawford — APPROVED** [review]
:+1: :cake:

**Pull request closed without merging**